src/reactphp/src/ServerFactory.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Runtime\React;
+
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use React\EventLoop\Loop;
+use React\Http\HttpServer;
+use React\Socket\SocketServer;
+
+class ServerFactory
+{
+    private const DEFAULT_OPTIONS = [
+        'host' => '127.0.0.1',
+        'port' => 8080,
+    ];
+    private array $options;
+
+    public static function getDefaultOptions(): array
+    {
+        return self::DEFAULT_OPTIONS;
+    }
+
+    public function __construct(array $options = [])
+    {
+        $options['host'] = $options['host'] ?? $_SERVER['REACT_HOST'] ?? $_ENV['REACT_HOST'] ?? self::DEFAULT_OPTIONS['host'];
+        $options['port'] = $options['port'] ?? $_SERVER['REACT_PORT'] ?? $_ENV['REACT_PORT'] ?? self::DEFAULT_OPTIONS['port'];
+
+        $this->options = array_replace_recursive(self::DEFAULT_OPTIONS, $options);
+    }
+
+    public function createServer(RequestHandlerInterface $requestHandler): HttpServer
+    {
+        $loop = Loop::get();
+        $server = new HttpServer($loop, function (ServerRequestInterface $request) use ($requestHandler) {
+            return $requestHandler->handle($request);
+        });
+
+        $socket = new SocketServer(sprintf('%s:%s', $this->options['host'], $this->options['port']), [], $loop);
+        $server->listen($socket);
+        $loop->run();
```

Comment:
I think the factory should not call run. Like the swoole factory does not call `start`. This should be  called in the runner.

Reply:
Done. I struggled with that when making the change - it looks a little strange now when you `createServer` but get a `Loop` in return, but that's how react does it (ie, you create an http server, register it with the event loop, then start the loop which has a side-effect of starting the server).

Reply:
Oh sorry didn't know that it is not called on the `server`. Maybe then it make sense to keep the `run` here. Or if there is possible to get the loop from the server if there is no `run` on the server. 

Else it would be a `LoopFactory::createLoop` but not sure if this make sense. I'm as said not very familiar with reactphp enough.

Maybe @Nyholm can give his opinion on this.

Reply:
We called $loop->run() twice. I removed this one and let the `Runner::run()` call `$loop->run()`